File: content/problems/0400-nth-digit/solutions/digit-block-location/Solution.kt
class Solution {
    fun findNthDigit(n:Int):Int{
        var position=n.toLong();
        var digit=1L;
        var start=1L;
        var count=9L;
        while(position>digit*count){
            position-=digit*count;
            digit++;
            start*=10;
            count*=10
        };
        val number=start+(position-1)/digit;
        return number.toString()[((position-1)%digit).toInt()]-'0'
    }
}

File: content/problems/0400-nth-digit/solutions/digit-block-location/solution.cpp
class Solution {
    public: int findNthDigit(int n){
        long long position=n,digit=1,start=1,count=9;
        while(position>digit*count){
            position-=digit*count;
            ++digit;
            start*=10;
            count*=10;
        }
        long long number=start+(position-1)/digit;
        return to_string(number)[(position-1)%digit]-'0';
    }
};
